DependencyInjection/Configuration.php

```diff
@@ -963,4 +1062,119 @@ public function getConfigTreeBuilder()
+        $acceptableParamsForHandlers = array_intersect($this->provideAllConfigurationParams(), array_keys($v));
+        $unacceptableParamForHandler = array_diff(
+            $acceptableParamsForHandlers,
+            $this->acceptedParamsByHandlerType[strtolower($v['type'])]
+        );
+        if (!empty($unacceptableParamForHandler)) {
+            throw new InvalidConfigurationException(sprintf(
+                'The handler type %s does not support %s %s',
+                strtolower($v['type']),
+                implode(', ', $unacceptableParamForHandler),
+                count($unacceptableParamForHandler) == 1 ? 'parameter' : 'parameters'
+            ));
+        }
+
+        return $v;
+    }
+
+    private function provideAllConfigurationParams()
+    {
+        return [
+            'accepted_levels',
+            'action_level',
+            'activation_strategy',
+            'api_version',
+            'app_name',
+            'auto_log_stacks',
+            'bubble',
+            'bot_name',
+            'buffer_size',
+            'channel',
+            'client_id',
+            'config',
+            'connection_string',
+            'connection_timeout',
+            'console_formater_options',
+            'content_type',
+            'date_format',
+            'deduplication_level',
+            'document_type',
+            'dsn',
+            'elasticsearch',
+            'email_prototype',
+            'environment',
+            'exchange',
+            'exchange_name',
+            'excluded_404s',
+            'excluded_http_codes',
+            'facility',
+            'filename_format',
+            'file_permission',
+            'flush_on_overflow',
+            'from_email',
+            'handler',
+            'headers',
+            'host',
+            'icon_emoji',
+            'id',
+            'ident',
+            'include_extra',
+            'index',
+            'lazy',
+            'level',
+            'logopts',
+            'mailer',
+            'max_files',
+            'max_level',
+            'message_format',
+            'message_type',
+            'min_level',
+            'members',
+            'mongo',
+            'nickname',
+            'notify',
+            'path',
+            'passthru_level',
+            'persistent',
+            'port',
+            'publisher',
+            'redis',
+            'release',
+            'region',
+            'room',
+            'source',
+            'stop_buffering',
+            'store',
+            'subject',
+            'tags',
+            'team',
+            'time',
+            'timeout',
+            'title',
+            'token',
+            'to_email',
+            'url',
+            'user',
+            'use_attachment',
+            'use_locking',
+            'use_short_attachment',
+            'use_ssl',
+            'verbosity_levels',
+            'webhook_url',
+        ];
```

Comment:
Could this be generated by recursively merging `$this->acceptedParamsByHandlerType` instead? So it doesn't need to be kept up to date, because I'm pretty sure it'll get outdated otherwise.